File: vokimi_api/Src/enums/TestTemplate.cs
namespace vokimi_api.Src.enums
{
    public enum TestTemplate
    {
        General,
        Scoring,
        CorrectAnswers
    }
    public static class TestTemplateExtensions
    {
        public static string GetId(this TestTemplate template) => template switch {
            TestTemplate.General => "general",
            TestTemplate.Scoring => "scoring",
            TestTemplate.CorrectAnswers => "correct_answers",
            _ => throw new NotImplementedException()
        };
        public static TestTemplate? FromId(string? id) => id switch {
            "general" => TestTemplate.General,
            "scoring" => TestTemplate.Scoring,
            "correct_answers" => TestTemplate.CorrectAnswers,
            _ => null
        };
    }
}

File: vokimi_frontend/src/ts/enums/TestTemplate.ts
export enum TestTemplate {
    General = "general",
    Scoring = "scoring",
    CorrectAnswers = "correct_answers"
}

export namespace TestTemplateUtils {
    export function fromId(id: string): TestTemplate {
        for (const key in TestTemplate) {
            if (TestTemplate[key as keyof typeof TestTemplate] === id) {
                return TestTemplate[key as keyof typeof TestTemplate];
            }
        }
        return TestTemplate.General;
    }

    export function getFullName(template: TestTemplate): string {
        switch (template) {
            case TestTemplate.General:
                return "General Test";
            case TestTemplate.Scoring:
                return "Scoring Test";
            case TestTemplate.CorrectAnswers:
                return "Correct Answers Test";
            default:
                return "Unknown Test Template";
        }
    }
    export function getTemplateFeatures(template: TestTemplate): string[] {
        switch (template) {
            case TestTemplate.General:
                return [
                    "Completely customize your test the way you want it",
                    "No restrictions or necessities (almost)",
                ];
            case TestTemplate.Scoring:
                return [
                    "Let the test takers see how well do they meet this or that criterion",
                    "Specially selected types of questions and the method of evaluating answers",
                ];
            case TestTemplate.CorrectAnswers:
                return ["", ""];
            default:
                throw new Error("Template not implemented");
        }
    }

}


